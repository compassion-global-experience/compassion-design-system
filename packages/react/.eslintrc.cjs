module.exports = {
  extends: [
    'standard-jsx',
    'standard-react',
    'plugin:jsx-a11y/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:storybook/recommended',
  ],
  env: {
    browser: true,
    es6: true,
    jest: true,
    node: false,
  },
  rules: {
    'jsx-quotes': ['error', 'prefer-double'],
  },
};
