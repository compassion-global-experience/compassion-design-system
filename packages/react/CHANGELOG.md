# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-04-08)

### Features

- **react:** css usage and documentation ([11e317c](https://github.com/compassion-global-experience/compassion-design-system/commit/11e317cca4ecdaa385e86d3a5571d7dca554685d))
- **react:** re-export css package ([67549a5](https://github.com/compassion-global-experience/compassion-design-system/commit/67549a58de3de6e02b58c5bdb6f4605001aae783))
- **react:** switch between themes in storybook ([79119ba](https://github.com/compassion-global-experience/compassion-design-system/commit/79119ba48ed3410eaebe9a91ccd076b05c2580c1))
