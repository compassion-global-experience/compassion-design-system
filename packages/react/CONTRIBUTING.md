# Contributing to the `react-web` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component‚Äôs structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component‚Äôs contents

## Workflow

The package uses [Storybook](https://storybook.js.org/docs/react/get-started/introduction)
to facilitate component development and documentation. Storybook uses the files
mentioned above to render components in their various states, and serves as
documentation of those components by surfacing code comments, `propTypes`, and
`defaultProps` values.

To launch Storybook, make sure you‚Äôre in the [/packages/react](/packages/react)
directory and run the command `npm run storybook`. After the project is built,
Storybook will launch in your default browser. New components or modifications
will be displayed in real time! üéâ

The configurations outlined in each component‚Äôs .stories.ts file can be seen in
Storybook‚Äôs **Docs** tab.

## Commit Messages & Versioning

Every package in the compassion-gds monorepo adheres to
[semantic versioning](https://semver.org/). In order to simplify the versioning
and release process, please be sure your commit messages adhere to the
[‚ÄúConventional Commits‚Äù pattern](https://www.conventionalcommits.org/en/v1.0.0/#summary).
You may wish to install the [commitizen/cz-cli](https://github.com/commitizen/cz-cli)
to make this near-bulletproof.

## Making a Pull Request

Once your changes are ready, open a pull request against the `development`
branch of the project, and it will be merged in as quickly as possible!
