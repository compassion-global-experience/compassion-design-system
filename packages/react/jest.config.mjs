/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/configuration
 */

export default {
  // A preset that is used as a base for Jest's configuration
  preset: 'ts-jest/presets/default-esm',

  // Automatically clear mock calls, instances, contexts and results before every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // Indicates which provider should be used to instrument code for coverage
  coverageProvider: "v8",

  // The test environment that will be used for testing
  testEnvironment: "jsdom",

  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
  moduleNameMapper: {
    ".(css|less|scss)$": "identity-obj-proxy",
  },

  // Whether to use watchman for file crawling
  watchman: true,
};
