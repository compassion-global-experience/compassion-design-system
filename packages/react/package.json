{
  "name": "@compassion-gds/react",
  "version": "0.2.0-alpha.2",
  "description": "",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "browser": {
        "default": "./dist/esm/index.js"
      },
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
    },
    "./css/base.css": "./dist/css/reset.css",
    "./css/light.css": "./dist/css/cds-light.css",
    "./css/dark.css": "./dist/css/cds-dark.css",
    "./css/*": "./dist/css/*"
  },
  "scripts": {
    "dev": "rollup -c -w",
    "copy-css": "bash ./scripts/copy-css.sh",
    "prepare": "rollup -c && npm run copy-css",
    "test": "jest",
    "storybook": "storybook dev -p 6007",
    "build:storybook": "storybook build --quiet",
    "percy": "cross-env PERCY_TOKEN=$PERCY_TOKEN_REACT percy storybook ./storybook-static",
    "percy-snapshots": "npm run build:storybook && npm run percy"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/compassion-global-experience/compassion-design-system.git",
    "directory": "packages/react"
  },
  "keywords": [
    "compassion",
    "compassion-gds",
    "gds",
    "react"
  ],
  "author": "Compassion XD",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/compassion-global-experience/compassion-design-system/issues"
  },
  "homepage": "https://github.com/compassion-global-experience/compassion-design-system#readme",
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@compassion-gds/core": "^0.1.0-alpha.0",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-typescript": "^11.1.0",
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-docs": "^7.0.2",
    "@storybook/addon-essentials": "^7.0.2",
    "@storybook/addon-interactions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/preset-react-webpack": "7.0.2",
    "@storybook/react": "7.0.2",
    "@storybook/react-webpack5": "7.0.2",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.27",
    "babel-loader": "^8.3.0",
    "eslint": "^8.33.0",
    "eslint-config-standard-jsx": "^11.0.0",
    "eslint-config-standard-react": "^13.0.0",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-storybook": "^0.6.10",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "phosphor-react": "^1.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.20.2",
    "rollup-plugin-dts": "^5.1.1",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "storybook": "7.0.2",
    "ts-jest": "^29.0.5",
    "webpack": "^5.75.0"
  },
  "peerDependencies": {
    "react": "16.8.x || 17.x || 18.x",
    "react-dom": "16.8.x || 17.x || 18.x"
  }
}
