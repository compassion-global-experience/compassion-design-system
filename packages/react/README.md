# GDS React Components

The Compassion Design System includes a streamlined set of React components for
fast, accessible development of apps and static sites.

## Installing and usage

To install the GDS React package use

```shell
npm install @compassion-gds/react
```

Or `yarn add @compassion-gds/react` if you're using yarn

Import the following css files from the entry point of your application

#### index.js
```js
// Replace {theme} with any of the available themes (e.g. light, dark)
import '@compassion-gds/react/css/base.css';
import '@compassion-gds/react/css/vars/{theme}.css';
```

GDS React components would follow the imported theme.

Sample component usage:

```jsx
import React from 'react';
import { Button } from '@compassion-gds/react';

export const PrimaryButton = (props) => (
  <Button primary size="small" {...props} />
);
```

## Themes

Available options:
- light: `@compassion-gds/react/css/light.css`
- dark: `@compassion-gds/react/css/dark.css`

## Contributing to the `react` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component’s structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component’s contents
