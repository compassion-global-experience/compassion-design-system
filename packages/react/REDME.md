# GDS React Components

The Compassion Design System includes a streamlined set of React components for
fast, accessible development of apps and static sites.

## Installing and usage

To install the GDS React package use

```shell
npm install @compassion-gds/react
```

Or `yarn add @compassion-gds/react` if you're using yarn  
(`@compassion-gds/css` is automatically installed as peer dependency)

Import the following css files from the entry point of your application

#### index.js
```js
// Replace {theme} with any of the available themes (e.g. light, dark)
import '@compassion-gds/css/reset.css';
import '@compassion-gds/css/vars/{theme}.css';
```

GDS React components would follow the imported theme.

Sample component usage:

```jsx
import React from 'react';
import { Button } from '@compassion-gds/react';

export const PrimaryButton = (props) => (
  <Button primary size="small" {...props} />
);
```

## Themes

Available options:
- light: `@compassion-gds/css/vars/light.css`
- dark: `@compassion-gds/css/vars/dark.css`
