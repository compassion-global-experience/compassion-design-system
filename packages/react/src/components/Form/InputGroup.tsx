import { forwardRef } from 'react';

import Input from './Input';
import { TextFieldProps } from './TextField';
import { InputGroupContainer, AdornmentProps } from './Helpers';

export interface InputGroupProps extends TextFieldProps, AdornmentProps {}

const InputGroup = forwardRef<HTMLInputElement, InputGroupProps>(
  (props, ref) => {
    const {
      id = '',
      label = '',
      state,
      size,
      hint = '',
      adornmentPosition = 'end',
      adornmentText = '',
      adornmentIcon,
      className,
      fieldClassName,
      ...rest
    } = props;

    return (
      <InputGroupContainer
        id={id}
        label={label}
        state={state}
        size={size}
        hint={hint}
        className={className}
        adornmentPosition={adornmentPosition}
        adornmentText={adornmentText}
        adornmentIcon={adornmentIcon}
      >
        <Input
          ref={ref}
          id={id}
          state={state}
          size={size}
          className={fieldClassName}
          {...rest}
        />
      </InputGroupContainer>
    );
  },
);

InputGroup.displayName = 'InputGroup';

export default InputGroup;
