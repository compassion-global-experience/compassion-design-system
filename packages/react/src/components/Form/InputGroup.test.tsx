import { render } from '@testing-library/react';

import InputGroup from './InputGroup';

describe('InputGroup', () => {
  it('should render `Small` size', () => {
    const { container } = render(<InputGroup id="small-field" label="Small" size="small" />);
    expect(container).toMatchSnapshot();
  });

  it('should render `Success` state with `Large` size', () => {
    const { container } = render(<InputGroup id="success-field" label="Success" state="success" />);
    expect(container).toMatchSnapshot();
  });

  it('should render `Error` state', () => {
    const { container } = render(<InputGroup id="error-field" label="Error" state="error" />);
    expect(container).toMatchSnapshot();
  });

  it('should render `Disabled` state', () => {
    const { container } = render(<InputGroup id="disabled-field" label="Disabled" state="disabled" />);
    expect(container).toMatchSnapshot();
  });

  it('should render left aligned adornment with text', () => {
    const { container } = render(<InputGroup id="adorned-field" label="Adorned" adornmentPosition="start" adornmentText="Text" />);
    expect(container).toMatchSnapshot();
  });
});
