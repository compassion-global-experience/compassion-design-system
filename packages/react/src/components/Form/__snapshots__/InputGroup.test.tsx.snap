// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render \`Disabled\` state 1`] = `
<div>
  <div
    class="form-field-row "
  >
    <label
      class="form-label disabled"
      for="disabled-field"
    >
      Disabled
    </label>
    <div
      class="input-group"
    >
      <div
        class="form-field disabled medium "
      >
        <input
          class="form-input"
          disabled=""
          id="disabled-field"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group-adornment "
      >
        <svg
          fill="currentColor"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="120 120 128 120 128 176 136 176"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="126"
            cy="84"
            r="12"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`InputGroup should render \`Error\` state 1`] = `
<div>
  <div
    class="form-field-row "
  >
    <label
      class="form-label error"
      for="error-field"
    >
      Error
    </label>
    <div
      class="input-group"
    >
      <div
        class="form-field error medium "
      >
        <input
          class="form-input"
          id="error-field"
          placeholder=""
          type="text"
          value=""
        />
        <span
          class="form-icon"
        >
          <svg
            fill="currentColor"
            height="20"
            viewBox="0 0 256 256"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="200"
              x2="56"
              y1="56"
              y2="200"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="200"
              x2="56"
              y1="200"
              y2="56"
            />
          </svg>
        </span>
      </div>
      <div
        class="input-group-adornment "
      >
        <svg
          fill="currentColor"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="120 120 128 120 128 176 136 176"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="126"
            cy="84"
            r="12"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`InputGroup should render \`Small\` size 1`] = `
<div>
  <div
    class="form-field-row "
  >
    <label
      class="form-label "
      for="small-field"
    >
      Small
    </label>
    <div
      class="input-group"
    >
      <div
        class="form-field  small "
      >
        <input
          class="form-input"
          id="small-field"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group-adornment small"
      >
        <svg
          fill="currentColor"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="120 120 128 120 128 176 136 176"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="126"
            cy="84"
            r="12"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`InputGroup should render \`Success\` state with \`Large\` size 1`] = `
<div>
  <div
    class="form-field-row "
  >
    <label
      class="form-label success"
      for="success-field"
    >
      Success
    </label>
    <div
      class="input-group"
    >
      <div
        class="form-field success medium "
      >
        <input
          class="form-input"
          id="success-field"
          placeholder=""
          type="text"
          value=""
        />
        <span
          class="form-icon"
        >
          <svg
            fill="currentColor"
            height="20"
            viewBox="0 0 256 256"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <polyline
              fill="none"
              points="216 72 104 184 48 128"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </span>
      </div>
      <div
        class="input-group-adornment "
      >
        <svg
          fill="currentColor"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="120 120 128 120 128 176 136 176"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="126"
            cy="84"
            r="12"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`InputGroup should render left aligned adornment with text 1`] = `
<div>
  <div
    class="form-field-row "
  >
    <label
      class="form-label "
      for="adorned-field"
    >
      Adorned
    </label>
    <div
      class="input-group"
    >
      <div
        class="input-group-adornment "
      >
        <svg
          fill="currentColor"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="120 120 128 120 128 176 136 176"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="126"
            cy="84"
            r="12"
          />
        </svg>
        <span
          class="input-group-adornment-text"
        >
          Text
        </span>
      </div>
      <div
        class="form-field  medium "
      >
        <input
          class="form-input"
          id="adorned-field"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
